resolver-dns/src/main/java/io/netty/resolver/dns/DnsNameResolver.java

```diff
@@ -931,4 +934,12 @@ public void exceptionCaught(ChannelHandlerContext ctx, Throwable cause) throws E
             logger.warn("{} Unexpected exception: ", ch, cause);
         }
     }
+
+    @Override
+    protected InetSocketAddressResolver newInetSocketAddressResolver() {
+        HostsFileNameResolver hostsFileNameResolver = new HostsFileNameResolver(executor());
+        CompositeNameResolver<InetAddress> compositeResolver =
+                new CompositeNameResolver<InetAddress>(executor(), hostsFileNameResolver, this);
+        return new InetSocketAddressResolver(executor(), compositeResolver);
```

Comment:
Generally speaking, do we really want hosts file resolution to be a NameResolver?